File: appJava/src/main/java/com/brandon/appjava/utilities/NotificationDialog.java
package com.brandon.appjava.utilities;

import android.app.AlertDialog;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.TextView;

import com.brandon.appjava.R;

import java.util.Objects;
import java.util.function.Consumer;
import java.util.function.Supplier;

import javax.annotation.Nullable;

public class NotificationDialog {

    public void createNotification(Context context, String title, String body,@Nullable Runnable runnable) {
        final LayoutInflater inflater = LayoutInflater.from(context);
        final View view = inflater.inflate(R.layout.notification_dialog, null);
        final AlertDialog dialog = new AlertDialog.Builder(context).create();
        TextView titleText = view.findViewById(R.id.title_text);
        titleText.setText(title);
        TextView bodyText = view.findViewById(R.id.body_text);
        bodyText.setText(body);
        view.findViewById(R.id.dialog_dismiss_button).setOnClickListener(v -> {
            dialog.dismiss();
            Objects.requireNonNull(runnable).run();
        });
        dialog.show();
    }

}

File: legacyCore/src/main/java/com/brandon/uicore/NotificationDialog.kt
package com.brandon.uicore

import android.app.AlertDialog
import android.content.Context
import android.view.LayoutInflater
import android.widget.TextView
import com.brandon.legacycore.R

fun connectionDialog(context: Context, titleText: String, bodyText: String, onDismiss: () -> Unit = {}) {
    val inflater = LayoutInflater.from(context)
    val view = inflater.inflate(R.layout.notification_dialog, null)
    val dialog = AlertDialog.Builder(context).create()
    view.findViewById<TextView>(R.id.title_text).text = titleText
    view.findViewById<TextView>(R.id.body_text).text = bodyText
    view.findViewById<TextView>(R.id.dialog_dismiss_button).setOnClickListener {
        dialog.dismiss()
        onDismiss()
    }
    dialog.show()
}